src/systems/system.C

```diff
@@ -2144,6 +2144,38 @@ void System::user_QOI_derivative(const QoISet & qoi_indices,
       (qoi_indices, include_liftfunc, apply_constraints);
 }
 
+void System::init_qois(unsigned int n_qois)
+{
+  qoi.resize(n_qois);
+  qoi_error_estimates.resize(n_qois);
+}
+
+void System::fill_qoi(unsigned int qoi_index, Real qoi_value)
```

Comment:
Let's use "set" rather than "fill" for this and for the setter below?  More consistent with other library setters.